packages/block-editor/src/components/block-list/use-block-props/index.js

```diff
@@ -106,11 +106,12 @@ export function useBlockProps( props = {}, { __unstableIsHtml } = {} ) {
 	const isHovered = useIsHovered( ref );
 	const blockMovingModeClassNames = useBlockMovingModeClassNames( clientId );
 	const htmlSuffix = mode === 'html' && ! __unstableIsHtml ? '-visual' : '';
+	const mergedRefs = useMergeRefs( [ ref, useEventHandlers( clientId ) ] );
 
 	return {
 		...wrapperProps,
 		...props,
-		ref,
+		ref: mergedRefs,
```

Comment:
There are some places expecting the `ref` property to be a ref (not a callback). I made #29188 as one option to allow such expectations to be met.

The only alternative, I think, is to update consumers to pass callback refs to `useBlockProps` where needed. Though I'd guess it's desirable to avoid the breaking change if possible.